cloudinit/sources/helpers/azure.py

```diff
@@ -177,6 +185,46 @@ def report_diagnostic_event(str):
     return evt
 
 
+def report_compressed_event(event_name, event_content):
+    """Report a compressed event"""
+    compressed_data = base64.encodebytes(zlib.compress(event_content))
+    event_data = {"encoding": "gz+b64",
+                  "data": compressed_data.decode('ascii')}
+    evt = events.ReportingEvent(
+        COMPRESSED_EVENT_TYPE, event_name,
+        json.dumps(event_data),
+        events.DEFAULT_EVENT_ORIGIN)
+    events.report_event(evt, excluded_handlers={"logging"})
+
+    # return the event for unit testing purpose
+    return evt
+
+
+@azure_ds_telemetry_reporter
+def push_log_to_kvp(file_name=CFG_BUILTIN['def_log_file']):
```

Comment:
While this config is a sensible default. There may be (unlikely) custom azure images where that def_log_file is overridden. The easiest way to ensure you allways grab the image's configured def_log_file path would be through your DataSourceAzure.sys_cfg['def_log_file'] value.

We should pass 'self.sys_cfg['def_log_file']` into push_log_to_kvp where possible to any alterations to the default value from the merged /etc/cloud/cloud.cfg.d./*cfg files on disk